tests/rptest/tests/schema_registry_test.py

```diff
@@ -37,10 +41,30 @@ def create_topic_names(count):
 schema3_def = '{"type":"record","name":"myrecord","fields":[{"type":"string","name":"f1"},{"type":"string","name":"f2"}]}'
 
 
+def run_with_timeout(f, timeout=30):
+    class BackgroundThread(threading.Thread):
+        def __init__(self):
+            super(BackgroundThread, self).__init__()
+            self.result = None
+            self.complete = threading.Event()
+
+        def run(self):
+            self.result = f()
```

Comment:
why run the http requests in a background thread? it looks like its for the timeout, but it seemed like the request package provides that capability too.

Reply:
Yeah, this can go.  In the past I've had issues with the `requests` timeout detection being flaky, but once I got going using this test, the requests timeout was reliably catching issues where something got stuck.

Removed.